src/amazon/dsstne/engine/NNNetwork.cpp

```diff
@@ -3983,41 +3985,41 @@ bool NNNetwork::P2P_Allreduce(NNFloat* pBuffer, size_t size)
-            
+                uint64_t end                            = (size * (segment + 1)) / getGpu()._numprocs;
+
                 // Send segments around the adding local contributions from each process
                 for (uint32_t i = 0; i < stages; i++)
                 {
-               
+
                     if (i == 0)
                         cudaMemcpy(GetPeerBuffer(), pBuffer + start, (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                     else
-                        cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault); 
-                   
+                        cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
+
                     // Wait for completion
-                    cudaDeviceSynchronize();              
+                    cudaDeviceSynchronize();
                     MPI_Barrier(MPI_COMM_WORLD);
                     SwapPeerBuffers();
                     segment                             = (segment + 1) % getGpu()._numprocs;
                     start                               = (size * segment) / getGpu()._numprocs;
                     end                                 = (size * (segment + 1)) / getGpu()._numprocs;
-                    kAddBuffers(GetP2PSendBuffer(), pBuffer + start, end - start);  
+                    kAddBuffers(GetP2PSendBuffer(), pBuffer + start, end - start);
                 }
-                 
+
                 // Circulate segments a second time and copy out results
                 cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                 for (uint32_t i = 0; i < stages; i++)
                 {
-                    cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);   
-                   
-                    // Wait for completion  
-                    cudaDeviceSynchronize();              
+                    cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
+
+                    // Wait for completion
+                    cudaDeviceSynchronize();
                     MPI_Barrier(MPI_COMM_WORLD);
                     SwapPeerBuffers();
                     segment                             = (segment + 1) % getGpu()._numprocs;
                     start                               = (size * segment) / getGpu()._numprocs;
                     end                                 = (size * (segment + 1)) / getGpu()._numprocs;
-                    cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);               
+                    cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                 }
             }
         }
```

Comment:
General comment for the future CR:
It is better not to mix the whole file formatting with the real code change.
For example in this file, most of the changes related to code formatting and only couple of lines is real code change, so it would be better to split it into two CR: code formatting (and we do not need to review it) and real code change which has to be reviewed. Also code formatting can be done one for the whole code base, so we do not need to apply formatting file by file.